homeassistant/components/flexit_bacnet/sensor.py

```diff
@@ -0,0 +1,252 @@
+        native_unit_of_measurement=PERCENTAGE,
+        translation_key="heat_exchanger_efficiency",
+        value_fn=lambda data: data.heat_exchanger_efficiency,
+    ),
+    FlexitSensorEntityDescription(
+        key="heat_exchanger_speed",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+        translation_key="heat_exchanger_speed",
+        value_fn=lambda data: data.heat_exchanger_speed,
+    ),
+)
+
+# These attributes are already in the climate entity
+#   operation_mode
+#   ventilation_mode
+
+# Comfort button state: shows true or false if the hvac is in mode HOME or AWAY, and is probably not needed as a sensor
+# room humidity 1,2,3: gives readings if extra wireless hardware is installed (so not needed)
+# If the sensor with bacnet address 59 is non existent use 95 AND 96 for in machine humidity sensors
+
+# These attributes exist in flexit_bacnet but seem too redundant as sensors since they almost never change
+#   air_temp_setpoint_away
+#   air_temp_setpoint_home
+#   fan_setpoint_supply_air_home
+#   fan_setpoint_extract_air_home
+#   fan_setpoint_supply_air_high
+#   fan_setpoint_extract_air_high
+#   fan_setpoint_supply_air_away
+#   fan_setpoint_extract_air_away
+#   fan_setpoint_supply_air_cooker
+#   fan_setpoint_extract_air_cooker
+#   fan_setpoint_supply_air_fire
+#   fan_setpoint_extract_air_fire
+#   air_filter_exchange_interval
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Flexit (bacnet) sensor from a config entry."""
+    coordinator: FlexitCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    _LOGGER.info("Setting up Flexit (bacnet) sensor from a config entry")
+
+    async_add_entities(
+        [
+            FlexitSensor(coordinator, description, config_entry.entry_id)
+            for description in SENSOR_TYPES
+        ]
+    )
+
+    # TODO: unsubscribe on remove
+
+
+class FlexitSensor(FlexitEntity, CoordinatorEntity, SensorEntity):
+    """Representation of a Flexit Sensor."""
+
+    # Should it have a name?
+    # _attr_name = None
+
+    # Should it have a entity_name?
+    # _attr_has_entity_name = True
+
+    # Poll is default
+
+    entity_description: FlexitSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: FlexitCoordinator,
+        entity_description: FlexitSensorEntityDescription,
+        entry_id: str,
+    ) -> None:
+        """Initialize Flexit (bacnet) sensor."""
+        super().__init__(coordinator)
+
+        _LOGGER.info("Initialize Flexit (bacnet) sensor %s", entity_description.key)
+
+        self.entity_description = entity_description
+        self.entity_id = ENTITY_ID_SENSOR_FORMAT.format(
+            coordinator.device.device_name, entity_description.key
+        )
+        self._attr_unique_id = f"{entry_id}-{entity_description.key}"
```

Comment:
Instead of using an entry_id for the unique_id, can we use the serialnumber? The idea of unique_ids is that they should be unique for this machine. If we keep the config_entry.entry_id as unique_id, all automations would break if someone set up the integration again. We only like to do this when we actually can't do anything other. In this case we have a static serial_number we can use.

Reply:
It took me a while to understand this, but finally I got it.

Fixed!